File: LeetCode/1010. Pairs of Songs With Total Durations Divisible by 60.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        int m[60] = {} ;
        int ans = 0 ;
        for(auto x:time){
            int t = x % 60;
            int y = (60 - t)%60;
            ans += m[y];
            m[t]++;
        }
        return ans;
    }
};

File: LeetCode/1010. Pairs of Songs With Total Durations Divisible by 60.py
###  KNOWLEDGE IS POWER   ###

class Solution(object):
    def numPairsDivisibleBy60(self, time):
        """
        :type time: List[int]
        :rtype: int
        """
        m = [0] * 60
        ans = 0
        for x in time:
            ans += m[-x % 60]
            m[x % 60] += 1
        return ans
